ui/desktop/tests/end2end/authentication.spec.js

```diff
@@ -0,0 +1,205 @@
+/* eslint-disable no-undef */
+const { test, expect } = require('@playwright/test');
+const { _electron: electron } = require('playwright');
+const path = require('path');
+
+let electronApp = null;
+const returnScreenshotPath = (fileName) => {
+  const screenshotPath = 'tests/end2end/screenshots/authentication/';
+  const screenshotFormat = '.png';
+  return path.join(screenshotPath, fileName).concat(screenshotFormat);
+};
+
+test.beforeEach(async () => {
+  electronApp = await electron.launch({
+    executablePath:
+      'electron-app/out/Boundary-darwin-arm64/Boundary.app/Contents/MacOS/Boundary',
+    env: {
+      NODE_ENV: 'test',
+      BYPASS_APP_UPDATER: true,
+    },
+  });
+});
+
+test.afterEach(async () => {
+  electronApp.close();
+  electronApp = null; // Not sure we should do this.
+});
+
+test.describe('Authentication end to end test suite', async () => {
+  test.describe('user/password authentication test suite', async () => {
+    test('Authenticates using user and password method and deauthenticates ', async () => {
+      const boundaryWindow = await electronApp.firstWindow(); // The window that contains the app.
+      // Override local storage origin
+      await boundaryWindow.evaluate(() =>
+        window.localStorage.setItem('desktop:origin', null)
+      );
+      const originValue = 'http://localhost:9200';
+      const authLoginNameValue = 'admin';
+      const authLoginPasswordValue = 'password';
+
+      // Fill the origin input
+      await boundaryWindow.waitForSelector('.ember-text-field');
+      await boundaryWindow.fill('.ember-text-field', originValue);
+      // Take screenshot
+      await boundaryWindow.screenshot({
+        path: returnScreenshotPath('fillOrigin'),
+        fullPage: true,
+      });
+
+      // Click the submit button
+      // Due to an error with await boundaryWindow.click('button[type="submit"]'); we are using a workaround.
+      // More info about it here: https://github.com/microsoft/playwright/issues/1808
+      await boundaryWindow.waitForSelector('button[type="submit"]');
+      await boundaryWindow.$eval('button[type="submit"]', (element) =>
```

Comment:
Can we at least write our own function to handle click, to avoid duplicating this code?

Reply:
Yes! I am planning this on coming PR. That coming PR will focus on test helpers that can be shared between tests. 

So far we have next:
- The function that returns the path where te screenshot has to be saved.
- The function that returns where the Boundary Desktop client executable is stored. This change between platforms and archs.
- Our click function (as you suggested).

So I am going to tackle this in a separate PR